ocrd_modelfactory/ocrd_modelfactory/__init__.py

```diff
@@ -79,15 +82,32 @@ def page_from_image(input_file, with_tree=False):
 
 def page_from_file(input_file, with_tree=False):
     """
-    Create a new PAGE-XML from a METS file representing a PAGE-XML or an image.
+    Create :py:class:`~ocrd_models.ocrd_page.OcrdPage`
+    from an :py:class:`~ocrd_models.ocrd_file.OcrdFile` or a file path
+    representing either a PAGE-XML or an image (to generate a PAGE-XML for).
 
     Arguments:
-        input_file (:py:class:`~ocrd_models.ocrd_file.OcrdFile`): file to open \
+        input_file (:py:class:`~ocrd_models.ocrd_file.OcrdFile` or `str`): file to open \
             and produce a PAGE DOM for
     Keyword arguments:
         with_tree (boolean): whether to return XML node tree, element-node mapping \
             and reverse mapping, too (cf. :py:func:`ocrd_models.ocrd_page.parseEtree`)
     """
+    if not isinstance(input_file, OcrdFile):
+        mimetype = guess(input_file)
+        if mimetype is None:
+            mimetype = guess_type(input_file)[0]
+        else:
+            mimetype = mimetype.mime
+        if mimetype is None:
+            mimetype = EXT_TO_MIME.get(Path(input_file).suffix, None)
```

Comment:
Perhaps we should move this to a utility function, also to document why we need the three mechanisms (`mimetype.guess`, `filetype.guess_type` and `EXT_TO_MIME`).

Reply:
Sure. But note that there's already a difference to what we need (regarding these detectors) in resmgr.download.

Reply:
Yes, but these differences (falling back to `application/octet-stream` vs. raising an error) can be handled with function args.

Reply:
https://github.com/OCR-D/core/pull/1045/commits/ed19b654d70eb9437ea1d46ce065c7471f08787e